modules/mauth-authenticator-akka-http/src/main/scala/com/mdsol/mauth/akka/http/MAuthDirectives.scala

```diff
@@ -112,6 +148,74 @@ trait MAuthDirectives extends StrictLogging {
       None
     }
   }
+
+  private def getQueryString(req: HttpRequest): String = req.uri.rawQueryString.getOrElse("")
+
+  private def extractRequestHeader(request: HttpRequest, headerName: String): String =
+    request.headers
+      .find(_.name().compareToIgnoreCase(headerName) == 0)
+      .map(_.value())
+      .getOrElse("")
```

Comment:
`request.getHeader(headerName).map(_.value()).orElse("")`

Reply:
updated as suggested

Reply:
Hmm after annotating the type it works for me: `request.getHeader(headerName).map[String](_.value()).orElse("")`

Reply:
Yeah,  this works 👍